### ISSUE TITLE
max_pool_with_argmax can gives different results on GPU than on CPU

### ISSUE THREAD

**Issue by jnd77** [tensorflow/tensorflow#33403]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): YES
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MacOS
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 1.13.1
- Python version: 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 10.1
- GPU model and memory: GPU 1080-Ti

**Describe the current behavior**

When running the test in the code below, one of the tests fail. Somehow, the argmax result of max_pool_with_argmax is different when running with self.session(use_gpu=True).
Interestingly, the failing test behaves as if it was ignoring the batch index in the argmax.

**Describe the expected behavior**

All 3 tests should pass.

**Code to reproduce the issue**

    import tensorflow as tf
    import numpy as np
    
    
    class TestMaxPool(tf.test.TestCase):

        def setUp(self):
            tf.reset_default_graph()

        def _test(self, sess):
             # Image with 2 channels, H = 4, W = 4
             image_t = tf.constant([[[1.0, 12.0], [3.0, 14.0], [5.0, 16.0], [7.0, 18.0]],
                               [[11.0, 2.0], [13.0, 4.0], [15.0, 6.0], [17.0, 8.0]],
                               [[38.0, 37.0], [36.0, 35.0], [34.0, 33.0], [32.0, 31.0]],
                               [[28.0, 27.0], [26.0, 25.0], [24.0, 23.0], [22.0, 21.0]]])
             input_t = tf.stack([image_t, image_t])
             _, pool_indices = tf.nn.max_pool_with_argmax(input_t, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1], padding='SAME')
             pool_indices_out = sess.run(pool_indices)

            expected_image_indices = np.array([[[10, 3], [14, 7]], [[16, 17], [20, 21]]])
            self.assertAllEqual(pool_indices_out, np.stack([expected_image_indices, expected_image_indices + 32]))

        def test_with_normal_session(self):
            with tf.Session() as sess:
                self._test(sess)

        def test_with_self_session_and_gpu(self):
            with self.session(use_gpu=True) as sess:
                self._test(sess)

        def test_with_self_session_and_cpu(self):
            with self.session(use_gpu=False) as sess:
                self._test(sess)

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by gadagashwini** [comment]
@jnd77, Thanks for reporting the issue.
Please provide the complete code to reproduce the reported issue. Thanks!

**Comment by gadagashwini** [comment]
@jnd77, I tried replicating the reported issue. Please take a look at the [gist](https://colab.sandbox.google.com/gist/gadagashwini/696ae88444b31f485f5a8a97c723bfb5/untitled210.ipynb). And provide the more information to reproduce the issue. Thanks!

**Comment by jnd77** [comment]
@gadagashwini . Here is the [gist ](https://colab.research.google.com/gist/jnd77/e7193cb2eb57c23a35e460c6538751ff/untitled210.ipynb) reproducing the issue

**Comment by jvishnuvardhan** [comment]
@jnd77 Is this still an issue? Can you please try with `TF1.15` and let us know whether the issue persists. If possible, test it in `tf-nightly` or 'TF2.1`. Thanks!

**Comment by jnd77** [comment]
@jvishnuvardhan I did a check with TF1.15.
The function has this new argument: include_batch_in_index
Both CPU and GPU versions give the same (correct) result, whether include_batch_in_index is True or False. And documentation has been updated as well.

All good on my side ! :)
